### PR TITLE
Feature/mds 2349

### PR DESCRIPTION
**Pull request by mmumby**

### CODE REVIEW

File: migrations/sql/V2019.10.10.15.47__add_now_party_appointment.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    FOREIGN KEY (mine_party_appt_type_code) REFERENCES mine_commodity_code(mine_party_appt_type_code) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (application_id) REFERENCES application(id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (party_guid) REFERENCES party(party_guid) DEFERRABLE INITIALLY DEFERRED
+);
+
+alter table party ADD COLUMN bc_federal_incorporation_number varchar(30);
+alter table party ADD COLUMN bc_registration_number varchar(10);
+alter table party ADD COLUMN society_number varchar(30);
+alter table party ADD COLUMN hst_registration_number varchar(30);
```
Comment (nayr974) [inline migrations/sql/V2019.10.10.15.47__add_now_party_appointment.sql]: should we change this to tax_registration_number?

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+  reclamation_description character varying(4000),
+  reclamation_cost numeric(10,2),
+  total_disturbed_area numeric(14,2),
+  total_disturbed_area_unit_type_code character varying(3),
+  create_user character varying(60) NOT NULL,
+  create_timestamp timestamp with time zone DEFAULT now() NOT NULL,
+  update_user character varying(60) NOT NULL,
+  update_timestamp timestamp with time zone DEFAULT now() NOT NULL,
+
+  FOREIGN KEY (permit_application_id) REFERENCES permit_application(permit_application_id)
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: Don't make any changes, but if you put "REFERENCES ...." directly on the column creation, you don't need to specify FOREIGN KEY and it saves a bit of typing.

File: migrations/sql/V2019.10.10.15.54__create_application_document_tables.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    active_ind                       boolean DEFAULT true NOT NULL,
+    create_user                      character varying(60) NOT NULL,
+    create_timestamp                 timestamp with time zone DEFAULT now() NOT NULL,
+    update_user                      character varying(60) NOT NULL,
+    update_timestamp                 timestamp with time zone DEFAULT now() NOT NULL
+);
+
+ALTER TABLE application_document_type OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS application_document_xref
```
Comment (nayr974) [inline migrations/sql/V2019.10.10.15.54__create_application_document_tables.sql]: Ryan should have added table comments.

File: migrations/sql/V2019.10.10.15.54__create_application_document_tables.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+CREATE TABLE IF NOT EXISTS application_document_type (
```
Comment (nayr974) [inline migrations/sql/V2019.10.10.15.54__create_application_document_tables.sql]: Ryan should have added table comments.

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+CREATE TABLE settling_pond (
+  settling_pond_id SERIAL PRIMARY KEY,
+  proponent_pond_name character varying(4000),
+  water_source_description character varying(4000),
+  total_disturbed_area numeric(14,2),
+  total_disturbed_unit_type_code character varying(3),
+  reclamation_description character varying(4000),
+  reclamation_cost numeric(10,2),
+  construction_plan character varying(4000),
+  is_ponds_exfiltrated boolean,
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: is_or_sometimes_has_xxxx_created_by_md naming convention, _please_.

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+
+
+CREATE TABLE exploration_surface_drilling (
+  exploration_surface_drilling_id SERIAL PRIMARY KEY,
+  permit_application_id integer,
+  reclamation_description character varying(4000),
+  reclamation_core_storage character varying(4000),
+  reclamation_cost numeric(10,2),
+  total_disturbed_area numeric(14,2),
+  total_disturbed_area_unit_type_code character varying(3),
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: needs FK to units

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 7]
```diff
@@ -0,0 +1,525 @@
...
+COMMENT ON TABLE exploration_surface_drilling IS 'A list related to a Notice of Work activity - Exploration Surface Drilling';
+
+
+CREATE TABLE sand_gravel_quarry_operation (
+  sand_gravel_quarry_operation_id SERIAL PRIMARY KEY,
+  permit_application_id integer,
+  average_overburden_depth numeric(14,2),
+  average_top_soil_depth numeric(14,2),
+  stability_measures_description character varying(4000),
+  is_agricultural_land_reserve bolean,
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: boolean vs bolean

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 8]
```diff
@@ -0,0 +1,525 @@
...
+  DEFERRABLE INITIALLY DEFERRED,
+);
+
+COMMENT ON TABLE underground_exploration IS 'A list related to a Notice of Work activity - Underground Exploration';
+
+CREATE TABLE camp (
+  camp_id SERIAL PRIMARY KEY,
+  permit_application_id integer,
+  camp_name character varying,
+  camp_nunmber_people character varying,
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: number

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql (hunk 5 above)
Comment (mmumby) [reply migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: lolz

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 9]
```diff
@@ -0,0 +1,525 @@
...
+  active_ind boolean,
+  create_user character varying(60) NOT NULL,
+  create_timestamp timestamp with time zone DEFAULT now() NOT NULL,
+  update_user character varying(60) NOT NULL,
+  update_timestamp timestamp with time zone DEFAULT now() NOT NULL
+);
+
+COMMENT ON TABLE notice_of_work_type_code IS 'A code table containing the notice of work type codes and values';
+
+CREATE TABLE application_ststus (
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: status

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql [hunk 10]
```diff
@@ -0,0 +1,525 @@
...
+  application_status_code character varying(3),
+  description character varying(100),
+  active_ind boolean,
+  create_user character varying(60) NOT NULL,
+  create_timestamp timestamp with time zone DEFAULT now() NOT NULL,
+  update_user character varying(60) NOT NULL,
+  update_timestamp timestamp with time zone DEFAULT now() NOT NULL
+);
+
+COMMENT ON TABLE application_ststus_code IS 'A code table containing the application status codes and values';
```
Comment (nayr974) [inline migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: status

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql (hunk 6 above)
Comment (mmumby) [reply migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: added!

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql (hunk 7 above)
Comment (mmumby) [reply migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: fixed

File: migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql (hunk 8 above)
Comment (mmumby) [reply migrations/sql/V2019.10.11.09.53_create_notice_of_work_information.sql]: fixed

File: migrations/sql/V2019.10.10.15.47__add_now_party_appointment.sql (hunk 1 above)
Comment (JustinMacaulay) [reply migrations/sql/V2019.10.10.15.47__add_now_party_appointment.sql]: Yeah that probably makes more sense, I'll change it.

**Pull request merged**